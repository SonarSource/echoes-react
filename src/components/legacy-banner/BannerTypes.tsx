/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { TextNodeOptional } from '~types/utils';
import { BannerVariety } from '../layout';

export interface LegacyBannerProps {
  /**
   * The content to be displayed in the banner, keep it short and concise. It can't break into multiple lines and will be ellipsized if too long.
   */
  children: TextNodeOptional;
  className?: string;
  /**
   * Set to true to disable the follow horizontal scroll behavior of the banner.
   */
  disableFollowScroll?: boolean;
  /**
   * Function that will be called when the dismiss button is clicked, the dismiss button is only showed if this function is defined.
   */
  onDismiss?: VoidFunction | ((event: React.MouseEvent<HTMLButtonElement>) => Promise<void>);
  /**
   * Prefix text for screen readers to announce before the banner content. Optional since a default value is provided.
   */
  screenReaderPrefix?: TextNodeOptional;
  /**
   * The variety of banner to display, controls the styling and icon.
   */
  variety: `${BannerVariety}`;
  /**
   * Custom z-index for the banner, defaults to 1.
   */
  zIndex?: number;
}
