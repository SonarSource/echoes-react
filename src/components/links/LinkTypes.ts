/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import React from 'react';
import { LinkProps as RouterLinkProps } from 'react-router-dom';

type RouterNavLinkPropsAllowed = 'download' | 'reloadDocument' | 'state' | 'style' | 'title' | 'to';

export enum LinkHighlight {
  Accent = 'accent',
  CurrentColor = 'current-color',
  Default = 'default',
  Subdued = 'subdued',
}

export interface LinkProps extends Pick<RouterLinkProps, RouterNavLinkPropsAllowed> {
  children: React.ReactNode;
  className?: string;
  highlight?: `${LinkHighlight}`;
  onClick?: (event: React.MouseEvent<HTMLAnchorElement>) => void;
  shouldBlurAfterClick?: boolean;
  shouldOpenInNewTab?: boolean;
  shouldPreventDefault?: boolean;
  shouldStopPropagation?: boolean;
}
