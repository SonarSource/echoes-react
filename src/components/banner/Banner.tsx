/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { forwardRef, useMemo } from 'react';
import { useIntl } from 'react-intl';
import { ButtonIcon } from '../buttons';
import { IconX } from '../icons/IconX';
import { BannerScreenReaderPrefix } from './BannerScreenReaderPrefix';
import {
  BANNER_TYPE_ICONS,
  BANNER_TYPE_STYLES,
  BannerContent,
  BannerContentText,
  BannerInner,
  BannerSkeleton,
  BannerWrapper,
} from './BannerStyles';
import { BannerProps } from './BannerTypes';

export const Banner = forwardRef<HTMLDivElement, BannerProps>((props, ref) => {
  const { children, type, onDismiss, screenReaderPrefix, ...htmlProps } = props;
  const intl = useIntl();

  const { icon: BannerIcon, iconColor } = BANNER_TYPE_ICONS[type];

  return (
    <BannerSkeleton css={useMemo(() => BANNER_TYPE_STYLES[type], [type])} role="alert">
      <BannerWrapper>
        <BannerInner ref={ref} {...htmlProps}>
          <BannerContent>
            <BannerIcon color={iconColor} />
            <BannerScreenReaderPrefix screenReaderPrefix={screenReaderPrefix} type={type} />
            <BannerContentText>{children}</BannerContentText>
          </BannerContent>

          {onDismiss && (
            <ButtonIcon
              Icon={IconX}
              ariaLabel={intl.formatMessage({
                id: 'banner.dismiss',
                defaultMessage: 'Dismiss',
                description:
                  'ARIA-label for the dismiss button in the top-right corner of the Banner.',
              })}
              onClick={onDismiss}
              size="medium"
              variety="default-ghost"
            />
          )}
        </BannerInner>
      </BannerWrapper>
    </BannerSkeleton>
  );
});

Banner.displayName = 'Banner';
