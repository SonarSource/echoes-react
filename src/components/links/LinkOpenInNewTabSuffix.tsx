/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import * as VisuallyHidden from '@radix-ui/react-visually-hidden';
import { FormattedMessage } from 'react-intl';
import { IconLinkExternal } from '../icons/IconLinkExternal';
import { LinkBaseProps } from './LinkTypes';

type Props = Pick<LinkBaseProps, 'enableOpenInNewTab'> & {
  hasUnbreakableSpace?: boolean;
};

export function LinkOpenInNewTabSuffix(props: Readonly<Props>) {
  const { enableOpenInNewTab, hasUnbreakableSpace = false } = props;

  if (!enableOpenInNewTab) {
    return null;
  }

  return (
    <>
      {hasUnbreakableSpace && <>&nbsp;</>}
      <IconLinkExternal data-testid="echoes-link-external-icon" />
      <VisuallyHidden.Root>
        <FormattedMessage
          defaultMessage="(opens in new tab)"
          description="Screen reader-only text to indicate that the link will open in a new tab"
          id="open_in_new_tab"
        />
      </VisuallyHidden.Root>
    </>
  );
}

LinkOpenInNewTabSuffix.displayName = 'LinkOpenInNewTabSuffix';
