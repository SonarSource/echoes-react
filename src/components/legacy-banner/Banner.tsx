/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { forwardRef, useMemo } from 'react';
import { useIntl } from 'react-intl';
import { DismissButton } from '~common/components/DismissButton';
import { useFollowScroll } from '~common/helpers/useFollowScroll';
import { BannerScreenReaderPrefix } from '../layout/banner/BannerScreenReaderPrefix';
import {
  BANNER_TYPE_ICONS,
  BANNER_TYPE_STYLES,
  BannerContent,
  BannerContentText,
} from '../layout/banner/BannerStyles';
import { BannerInner, BannerSkeleton, BannerWrapper } from './BannerStyles';
import { LegacyBannerProps } from './BannerTypes';

/**
 * @deprecated Legacy Banner, use the new Layout.Banner instead. If your Banner is not displayed
 * within the Layout.BannerContainer, then it should most probably be migrated to a MessageCallout instead.
 */
export const Banner = forwardRef<HTMLDivElement, LegacyBannerProps>((props, ref) => {
  const {
    children,
    disableFollowScroll = false,
    onDismiss,
    screenReaderPrefix,
    variety,
    zIndex = 1,
    ...htmlProps
  } = props;

  const intl = useIntl();
  const { left: leftScroll } = useFollowScroll(!disableFollowScroll);

  const wrapperLeftOffset = disableFollowScroll ? {} : { left: -leftScroll };

  const { icon: BannerIcon, iconColor } = BANNER_TYPE_ICONS[variety];

  return (
    <BannerSkeleton
      css={useMemo(
        () => ({
          ...BANNER_TYPE_STYLES[variety],
          '--banner-z-index': zIndex,
        }),
        [variety, zIndex],
      )}
      role="alert">
      <BannerWrapper style={wrapperLeftOffset}>
        <BannerInner ref={ref} {...htmlProps}>
          <BannerContent>
            <BannerIcon color={iconColor} />
            <BannerScreenReaderPrefix screenReaderPrefix={screenReaderPrefix} variety={variety} />
            <BannerContentText>{children}</BannerContentText>
          </BannerContent>

          {onDismiss && (
            <DismissButton
              ariaLabel={intl.formatMessage({
                id: 'banner.dismiss',
                defaultMessage: 'Dismiss',
                description:
                  'ARIA-label for the dismiss button in the top-right corner of the Banner.',
              })}
              onClick={onDismiss}
            />
          )}
        </BannerInner>
      </BannerWrapper>
    </BannerSkeleton>
  );
});

Banner.displayName = 'LegacyBanner';
