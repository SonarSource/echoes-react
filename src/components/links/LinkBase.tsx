/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { ForwardedRef, forwardRef, MouseEvent, useCallback } from 'react';
import { Link as RouterLink } from 'react-router-dom';
import { isSonarLink } from '~common/helpers/url';
import { LinkOpenInNewTabSuffix } from './LinkOpenInNewTabSuffix';
import { isLinkAsButton, LinkProps } from './LinkTypes';

export const LinkBase = forwardRef<HTMLAnchorElement | HTMLButtonElement, LinkProps>(
  (props, ref) => {
    const {
      children,
      shouldBlurAfterClick = false,
      onClick,
      enablePreventDefault = false,
      shouldStopPropagation = false,
      enableOpenInNewTab = false,
      type = 'button',
      ...restProps
    } = props;

    const handleClick = useCallback(
      (event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => {
        if (shouldBlurAfterClick) {
          event.currentTarget.blur();
        }

        if (enablePreventDefault) {
          event.preventDefault();
        }

        if (shouldStopPropagation) {
          event.stopPropagation();
        }

        if (onClick) {
          onClick(event);
        }
      },
      [onClick, shouldBlurAfterClick, enablePreventDefault, shouldStopPropagation],
    );

    if (isLinkAsButton(props)) {
      return (
        <button
          {...restProps}
          onClick={handleClick}
          ref={ref as ForwardedRef<HTMLButtonElement>}
          // eslint-disable-next-line react/button-has-type
          type={type}>
          {children}
        </button>
      );
    }

    const { to } = props;

    return (
      <RouterLink
        {...getShouldOpenInNewTabProps({ enableOpenInNewTab, to })}
        {...restProps}
        onClick={handleClick}
        ref={ref as ForwardedRef<HTMLAnchorElement>}
        to={to}>
        {children}
        <LinkOpenInNewTabSuffix enableOpenInNewTab={enableOpenInNewTab} hasUnbreakableSpace />
      </RouterLink>
    );
  },
);

LinkBase.displayName = 'LinkBase';

export function getShouldOpenInNewTabProps({
  enableOpenInNewTab,
  to,
}: Pick<LinkProps, 'enableOpenInNewTab' | 'to'>) {
  return enableOpenInNewTab
    ? {
        rel: `noopener${typeof to === 'string' && isSonarLink(to) ? '' : ' noreferrer nofollow'}`,
        /* eslint-disable-next-line react/jsx-no-target-blank -- we only allow noopener noreferrer for known external links */
        target: '_blank',
      }
    : {};
}
