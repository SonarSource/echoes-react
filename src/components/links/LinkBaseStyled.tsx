/*
 * Echoes React
 * Copyright (C) 2023-2023 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { LinkBase } from './LinkBase';

export const LinkBaseStyled = styled(LinkBase)`
  color: var(--color);
  text-decoration-color: var(--color);
  text-decoration: var(--echoes-text-decoration-underline);

  &:visited {
    color: var(--color);
  }

  &:hover {
    color: var(--hover);
  }

  &:focus,
  &:active {
    color: var(--active);
    text-decoration-color: var(--active);
  }

  &:focus,
  &:focus-visible {
    outline: var(--echoes-color-border-focus) solid 2px;
    outline-offset: var(--echoes-dimension-space-25);
  }

  & > svg {
    vertical-align: text-bottom !important;
  }

  ${({ icon }) =>
    icon &&
    css`
      & > svg,
      & > img {
        margin-right: var(--echoes-dimension-space-50);
      }
    `};

  @media print {
    &,
    &:visited,
    &:hover,
    &:focus,
    &:active {
      color: 'currentColor';
      text-decoration: none;
    }
  }
`;
