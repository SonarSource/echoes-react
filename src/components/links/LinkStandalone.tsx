/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { css } from '@emotion/react';
import styled from '@emotion/styled';
import { forwardRef } from 'react';
import { LinkProps } from './LinkBase';
import { LinkBaseStyled } from './LinkBaseStyled';

interface Props extends LinkProps {
  iconLeft?: React.ReactNode;
}

const LinkStandaloneBase = forwardRef<HTMLAnchorElement, Props>((props, ref) => {
  const { children, iconLeft, ...linkProps } = props;

  return (
    <LinkBaseStyled {...linkProps} ref={ref}>
      {iconLeft}

      {children}
    </LinkBaseStyled>
  );
});

LinkStandaloneBase.displayName = 'LinkStandaloneBase';

export const LinkStandalone = styled(LinkStandaloneBase)`
  text-decoration-line: var(--echoes-text-decoration-none);

  &:hover,
  &:focus,
  &:active {
    color: var(--color);
    text-decoration-color: var(--color);
  }

  &:hover {
    text-decoration-line: var(--echoes-text-decoration-underline);
  }

  ${({ iconLeft }) =>
    iconLeft &&
    css`
      & > svg,
      & > img {
        margin-right: var(--echoes-dimension-space-50);
      }
    `};
`;

LinkStandalone.displayName = 'LinkStandalone';
