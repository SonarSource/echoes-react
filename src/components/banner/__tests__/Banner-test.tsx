/*
 * Echoes React
 * Copyright (C) 2023-2025 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { screenReaderOnly } from '~common/helpers/styles';
import { render } from '~common/helpers/test-utils';
import { Link } from '../../links';
import { Banner } from '../Banner';
import { BannerProps, BannerType } from '../BannerTypes';

it('should display banner content', async () => {
  setupBanner({ children: 'Banner Content' });

  const banner = screen.getByRole('alert');
  expect(banner).toBeInTheDocument();
  expect(banner).toHaveTextContent('Information banner: Banner Content');
  expect(banner).toMatchSnapshot();

  await expect(banner).toHaveNoA11yViolations();
});

it.each([
  [BannerType.Danger, 'Error banner:'],
  [BannerType.Info, 'Information banner:'],
  [BannerType.Success, 'Success banner:'],
  [BannerType.Warning, 'Warning banner:'],
])('should render with the type %s and a default screenreader prefix', (type, expectedPrefix) => {
  setupBanner({ type });

  expect(screen.getByRole('alert')).toHaveTextContent(`${expectedPrefix} Default Banner Content`);

  expect(screen.getByText(expectedPrefix)).toHaveStyle(
    screenReaderOnly.styles.replace(/label:.*?;/, ''),
  );
});

it('should be dismissable', async () => {
  const onDismiss = jest.fn();
  const { container, user } = setupBanner({ onDismiss });

  await expect(container).toHaveNoA11yViolations();

  const dismissButton = screen.getByRole('button', { name: 'Dismiss' });
  expect(dismissButton).toBeInTheDocument();
  await user.click(dismissButton);

  expect(onDismiss).toHaveBeenCalled();
});

it('should not render dismiss button when onDismiss is not provided', () => {
  setupBanner();

  expect(screen.queryByRole('button', { name: 'Dismiss' })).not.toBeInTheDocument();
});

it('should add custom screen reader prefix when provided', () => {
  setupBanner({
    screenReaderPrefix: 'Important:',
    children: 'Banner Message',
  });

  const alertElement = screen.getByRole('alert');
  expect(alertElement).toHaveTextContent('Important: Banner Message');
});

it('should correctly support Links in the banner content', async () => {
  const { user } = render(
    <MemoryRouter initialEntries={['/initial']}>
      <Routes>
        <Route
          element={
            <Banner type={BannerType.Info}>
              Banner with <Link to="/test">link</Link>
            </Banner>
          }
          path="/initial"
        />
        <Route element={<>Test page</>} path="/test" />
      </Routes>
    </MemoryRouter>,
  );

  expect(screen.getByRole('link', { name: 'link' })).toBeInTheDocument();
  await user.click(screen.getByRole('link', { name: 'link' }));
  expect(screen.getByText('Test page')).toBeInTheDocument();
});

function setupBanner({ children, ...props }: Partial<BannerProps> = {}) {
  return render(
    <Banner type={BannerType.Info} {...props}>
      {children ?? 'Default Banner Content'}
    </Banner>,
  );
}
