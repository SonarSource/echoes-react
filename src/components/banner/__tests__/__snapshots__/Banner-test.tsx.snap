// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display banner content 1`] = `
.emotion-0 {
  height: var(--echoes-banner-sizes-height);
}

.emotion-2 {
  position: fixed;
  background-color: var(--echoes-color-background-default);
  height: inherit;
  width: 100%;
  min-width: var(--echoes-layout-sizes-min-width-default);
  max-width: var(--echoes-layout-sizes-max-width-full);
  z-index: var(--banner-z-index);
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  gap: var(--echoes-dimension-space-300);
  box-sizing: border-box;
  height: inherit;
  width: 100%;
  padding: var(--echoes-dimension-space-75) var(--echoes-dimension-space-200);
  overflow: hidden;
  background-color: var(--banner-background);
  color: var(--banner-color);
  font: var(--echoes-typography-text-default-regular);
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--echoes-dimension-space-200);
  width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-8 {
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  font-size: calc(1em + 4px);
  font-style: normal;
  font-weight: normal;
  height: calc(2em - 16px);
  line-height: calc(2em - 16px);
  text-align: center;
  vertical-align: bottom;
  width: calc(2em - 16px);
  color: var(--echoes-color-icon-info);
  font-variation-settings: 'FILL' 1;
  font-family: 'Material Symbols Rounded';
}

.emotion-10 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  clip: rect(0, 0, 0, 0);
  white-space: nowrap;
  border-width: 0;
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--echoes-dimension-space-50);
  padding: var(--echoes-dimension-space-50);
  margin-left: calc(-1 * var(--echoes-dimension-space-50));
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div
  class="emotion-0 emotion-1"
  role="alert"
>
  <div
    class="emotion-2 emotion-3"
    style="left: 0px;"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <div
        class="emotion-6 emotion-7"
      >
        <span
          aria-hidden="true"
          class="emotion-8 emotion-9"
        >
          î¢Ž
        </span>
        <span
          class="emotion-10 emotion-11"
        >
          Information banner:
           
        </span>
        <div
          class="emotion-12 emotion-13"
        >
          Banner Content
        </div>
      </div>
    </div>
  </div>
</div>
`;
